From 19e10627dc391e8a3a4a255f6cf226268dc76a01 Mon Sep 17 00:00:00 2001
From: Angelo Naselli <anaselli@linux.it>
Date: Sat, 25 May 2013 16:14:31 +0200
Subject: [PATCH 1001/1004] livecd-tools-18-dracut_conf

---
 imgcreate/live.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/imgcreate/live.py b/imgcreate/live.py
index 1f332dd..3fe4d3c 100755
--- a/imgcreate/live.py
+++ b/imgcreate/live.py
@@ -223,11 +223,11 @@ class LiveImageCreatorBase(LoopImageCreator):
         self.base_on = True
         LoopImageCreator._mount_instroot(self, base_on)
         self.__write_initrd_conf(self._instroot + "/etc/sysconfig/mkinitrd")
-        self.__write_dracut_conf(self._instroot + "/etc/dracut.conf.d/02livecd.conf")
+        self.__write_dracut_conf(self._instroot + "/etc/dracut.conf.d/99livecd.conf")
 
     def _unmount_instroot(self):
         self.__restore_file(self._instroot + "/etc/sysconfig/mkinitrd")
-        self.__restore_file(self._instroot + "/etc/dracut.conf.d/02livecd.conf")
+        self.__restore_file(self._instroot + "/etc/dracut.conf.d/99livecd.conf")
         LoopImageCreator._unmount_instroot(self)
 
     def __ensure_isodir(self):
-- 
2.13.5

