From e0be2af1d9ac0ad0dce1e37f61a1dea76f473902 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Fri, 25 Nov 2016 07:19:19 -0500
Subject: [PATCH 1003/1004] livecd-tools-24-extra_filesystems

---
 imgcreate/live.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imgcreate/live.py b/imgcreate/live.py
index cfa31ea..5843eb9 100755
--- a/imgcreate/live.py
+++ b/imgcreate/live.py
@@ -264,7 +264,7 @@ class LiveImageCreatorBase(LoopImageCreator):
         return env
 
     def __extra_filesystems(self):
-        return "vfat msdos isofs ext4 xfs btrfs";
+        return "vfat msdos isofs ext4 xfs btrfs squashfs";
 
     def __extra_drivers(self):
         retval = "sr_mod sd_mod ide-cd cdrom "
-- 
2.13.5

