From 6870b2982afc17b65d958cf187ca9e74d4f015d7 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Fri, 25 Nov 2016 09:06:29 -0500
Subject: [PATCH 1004/1004] livecd-tools-24-disable-efi

---
 imgcreate/live.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/imgcreate/live.py b/imgcreate/live.py
index 5843eb9..07333e6 100755
--- a/imgcreate/live.py
+++ b/imgcreate/live.py
@@ -253,7 +253,8 @@ class LiveImageCreatorBase(LoopImageCreator):
     def _create_bootconfig(self):
         """Configure the image so that it's bootable."""
         self._configure_bootloader(self.__ensure_isodir())
-        self._generate_efiboot(self.__ensure_isodir())
+        # TODO: Support EFI boot on Mageia
+        #self._generate_efiboot(self.__ensure_isodir())
 
     def _get_post_scripts_env(self, in_chroot):
         env = LoopImageCreator._get_post_scripts_env(self, in_chroot)
-- 
2.13.5

