Index: Makefile
===================================================================
--- Makefile
+++ Makefile	2009-01-15 14:45:38.000000000 +0100
@@ -11,9 +11,11 @@
 	python -c "import compileall as c; c.compile_dir('$(1)', force=1)"
 	python -O -c "import compileall as c; c.compile_dir('$(1)', force=1)"
 endef
-PYTHONDIR := $(shell python -c "import distutils.sysconfig as d; print d.get_python_lib()")
+LIBNAME := $(shell python -c "import sys as d; print d.lib")
+# TODO : Must fix python version
+PYTHONDIR := /usr/$(LIBNAME)/python2.7/site-packages/
 
-all: 
+all:
 
 man:
 	pod2man --section=8 --release="livecd-tools $(VERSION)" --center "LiveCD Tools" docs/livecd-creator.pod > docs/livecd-creator.8
