diff -Naur livecd-tools-18.8/imgcreate/live.py livecd-tools-18.8.tpg/imgcreate/live.py
--- livecd-tools-18.8/imgcreate/live.py	2014-03-28 20:28:36.000000000 +0000
+++ livecd-tools-18.8.tpg/imgcreate/live.py	2014-03-28 20:32:49.426835227 +0000
@@ -607,7 +607,7 @@
                                            liveargs = kern_opts,
                                            long = "Start " + long + " in basic graphics mode.",
                                            short = "basic" + index,
-                                           extra = "xdriver=vesa nomodeset",
+                                           extra = "nomodeset",
                                            help = "Try this option out if you're having trouble starting.",
                                            index = index))
             linux.append(self.__get_image_stanza(is_xen, isDracut,
@@ -669,6 +669,15 @@
   localboot 0x80
 """
 
+    def __get_grub2_stanza(self, isodir):
+        shutil.copyfile(self._instroot + "/isolinux/sgb.iso", isodir + "/isolinux/sgb.iso")
+        shutil.copyfile(self._instroot + "/isolinux/memdisk", isodir + "/isolinux/memdisk")
+        return """label Rescue
+  menu label Run super grub2 disk
+  kernel memdisk
+  append initrd=sgb.iso
+"""
+
     def _configure_syslinux_bootloader(self, isodir):
         """configure the boot loader"""
         makedirs(isodir + "/isolinux")
@@ -702,11 +709,16 @@
             cfg += b
             if c:
                 cfg += c
+#        for b in basic:
+#            cfg += b
+#        for c in check:
+#            cfg += c
 
         cfg += self.__get_memtest_stanza(isodir)
         cfg += "menu separator\n"
 
         cfg += self.__get_local_stanza(isodir)
+        cfg += self.__get_grub2_stanza(isodir)
         cfg += self._get_isolinux_stanzas(isodir)
 
         cfg += """menu separator
