diff -ur livecd-tools-031.orig/imgcreate/creator.py livecd-tools-031/imgcreate/creator.py
--- livecd-tools-031.orig/imgcreate/creator.py	2011-01-26 11:01:32.000000000 +0300
+++ livecd-tools-031/imgcreate/creator.py	2011-01-25 12:25:12.000000000 +0300
@@ -757,10 +757,10 @@
         kickstart.NetworkConfig(self._instroot).apply(ksh.network)
         kickstart.RPMMacroConfig(self._instroot).apply(self.ks)
 
-        self._create_bootconfig()
-
         self.__run_post_scripts()
 
+        self._create_bootconfig()
+
     def launch_shell(self):
         """Launch a shell in the install root.
 
diff -ur livecd-tools-031.orig/imgcreate/errors.py livecd-tools-031/imgcreate/errors.py
--- livecd-tools-031.orig/imgcreate/errors.py	2009-10-30 00:26:09.000000000 +0300
+++ livecd-tools-031/imgcreate/errors.py	2011-01-26 10:55:38.000000000 +0300
@@ -54,3 +54,5 @@
     pass
 class ResizeError(CreatorError):
     pass
+class InitramfsError(CreatorError):
+    pass
Только в livecd-tools-031/imgcreate: errors.py~
diff -ur livecd-tools-031.orig/imgcreate/live.py livecd-tools-031/imgcreate/live.py
--- livecd-tools-031.orig/imgcreate/live.py	2011-01-26 11:01:32.000000000 +0300
+++ livecd-tools-031/imgcreate/live.py	2011-01-26 10:59:15.000000000 +0300
@@ -385,14 +385,18 @@
         shutil.copyfile(bootdir + "/vmlinuz-" + version,
                         isodir + "/isolinux/vmlinuz" + index)
 
+        subprocess.call(["/sbin/dracut", "/boot/initramfs-" + version + ".img", version],
+                        preexec_fn = self._chroot)
+
         isDracut = False
         if os.path.exists(bootdir + "/initramfs-" + version + ".img"):
             shutil.copyfile(bootdir + "/initramfs-" + version + ".img",
                             isodir + "/isolinux/initrd" + index + ".img")
             isDracut = True
         else:
-            shutil.copyfile(bootdir + "/initrd-" + version + ".img",
-                            isodir + "/isolinux/initrd" + index + ".img")
+            raise InitramfsError("Unable to create initramfs-" + version + ".img")
+            #shutil.copyfile(bootdir + "/initrd-" + version + ".img",
+                            #isodir + "/isolinux/initrd" + index + ".img")
 
         is_xen = False
         if os.path.exists(bootdir + "/xen.gz-" + version[:-3]):
Только в livecd-tools-031/imgcreate: live.py~
Только в livecd-tools-031/imgcreate: live.py.orig
