diff -Naur livecd-tools-18.8/imgcreate/live.py livecd-tools-18.8.tpg/imgcreate/live.py
--- livecd-tools-18.8/imgcreate/live.py	2014-07-12 17:56:30.000000000 +0000
+++ livecd-tools-18.8.tpg/imgcreate/live.py	2014-07-12 17:58:34.960945528 +0000
@@ -114,9 +114,11 @@
         """
         r = kickstart.get_kernel_args(self.ks)
         if os.path.exists(self._instroot + "/usr/bin/rhgb"):
-            r += " rhgb vga=788 splash=silent logo.nologo"
+# (tpg) use vga=current for syslinux bug
+# https://issues.openmandriva.org/show_bug.cgi?id=869
+            r += " rhgb vga=current splash=silent logo.nologo"
         if os.path.exists(self._instroot + "/usr/bin/plymouth"):
-            r += " rhgb vga=788 splash=silent logo.nologo"
+            r += " rhgb vga=current splash=silent logo.nologo"
         return r
         
     def _get_mkisofs_options(self, isodir):
