diff -Nur livecd-tools-18.8.old/imgcreate/live.py livecd-tools-18.8/imgcreate/live.py
--- livecd-tools-18.8.old/imgcreate/live.py	2012-11-15 19:17:44.000000000 +0400
+++ livecd-tools-18.8/imgcreate/live.py	2012-11-15 20:00:50.266280910 +0400
@@ -698,14 +698,10 @@
   menu title Troubleshooting
 """
         # Add basic video and check to submenu
-#        for b, c in zip(basic, check):
-#            cfg += b
-#            if c:
-#                cfg += c
-        for b in basic:
+        for b, c in zip(basic, check):
             cfg += b
-        for c in check:
-            cfg += c
+            if c:
+                cfg += c
 
         cfg += self.__get_memtest_stanza(isodir)
         cfg += "menu separator\n"
