--- livecd-tools-18.8/imgcreate/live.py.omv~	2013-10-14 01:09:07.453833182 +0200
+++ livecd-tools-18.8/imgcreate/live.py	2013-10-14 01:12:41.749823217 +0200
@@ -1,4 +1,4 @@
-#	
+#
 # live.py : LiveImageCreator class for creating Live CD images
 #
 # Copyright 2007, Red Hat  Inc.
@@ -113,10 +113,13 @@ class LiveImageCreatorBase(LoopImageCrea
 
         """
         r = kickstart.get_kernel_args(self.ks)
+        # We use vga=0 by default because we want to use uvesafb rather than
+        # vesafb -- even if both are built
+        r += " vga=0"
         if os.path.exists(self._instroot + "/usr/bin/rhgb"):
-            r += " rhgb vga=788 splash=silent logo.nologo"
+            r += " rhgb uvesafb.mode_option=1024x768-16 splash=silent logo.nologo"
         if os.path.exists(self._instroot + "/usr/bin/plymouth"):
-            r += " rhgb vga=788 splash=silent logo.nologo"
+            r += " rhgb uvesafb.mode_option=1024x768-16 splash=silent logo.nologo"
         return r
         
     def _get_mkisofs_options(self, isodir):
