--- livecd-tools-031/imgcreate/fs.py.dmsetup	2011-06-28 15:17:57.000000000 -0300
+++ livecd-tools-031/imgcreate/fs.py	2011-06-28 15:18:03.000000000 -0300
@@ -552,7 +552,7 @@ class DeviceMapperSnapshot(object):
         # where C is the number of 512 byte sectors in use
         #
         try:
-            return int((out.split()[3]).split('/')[0]) * 512
+            return int((out.split()[4]).split('/')[0]) * 512
         except ValueError:
             raise SnapshotError("Failed to parse dmsetup status: " + out)
 
