--- livecd-tools-18.8/imgcreate/live.py.0009~	2013-10-21 20:06:15.737099063 +0200
+++ livecd-tools-18.8/imgcreate/live.py	2013-10-21 20:07:15.452352519 +0200
@@ -264,8 +264,8 @@ class LiveImageCreatorBase(LoopImageCrea
         retval = "sr_mod sd_mod ide-cd cdrom "
         for module in self.__modules:
             if module == "=usb":
-                retval = retval + "ehci_hcd uhci_hcd ohci_hcd "
-                retval = retval + "usb_storage usbhid "
+                retval = retval + "xhcd_hcd ehci_hcd ehci-pci ehci-platform uhci_hcd ohci_hcd "
+                retval = retval + "usb_storage evdev hid usbhid "
             elif module == "=firewire":
                 retval = retval + "firewire-sbp2 firewire-ohci "
                 retval = retval + "sbp2 ohci1394 ieee1394 "
