diff -Nur livecd-tools-031.orig/tools/livecd-iso-to-disk-oem.sh livecd-tools-031/tools/livecd-iso-to-disk-oem.sh
--- livecd-tools-031.orig/tools/livecd-iso-to-disk-oem.sh	2011-03-15 12:54:56.000000000 +0300
+++ livecd-tools-031/tools/livecd-iso-to-disk-oem.sh	2011-03-15 13:06:39.639928232 +0300
@@ -336,6 +336,26 @@
 
 }
 
+cp_p()
+{
+#add progress bar to copy process
+   strace -q -ewrite cp -- "${1}" "${2}" 2>&1 \
+      | awk '{
+        count += $NF
+            if (count % 10 == 0) {
+               percent = count / total_size * 100
+               printf "%3d%% [", percent
+               for (i=0;i<=percent;i++)
+                  printf "="
+               printf ">"
+               for (i=percent;i<100;i++)
+                  printf " "
+               printf "]\r"
+            }
+         }
+         END { print "" }' total_size=$(stat -c '%s' "${1}") count=0
+}
+ 
 cryptedhome=1
 keephome=1
 homesizemb=0
@@ -577,7 +597,7 @@
       cp $CDMNT/LiveOS/ext3fs.img $USBMNT/$LIVEOS/ext3fs.img || (umount $CDMNT ; exitclean)
       umount $CDMNT
   elif [ -f $CDMNT/LiveOS/squashfs.img ]; then
-      cp $CDMNT/LiveOS/squashfs.img $USBMNT/$LIVEOS/squashfs.img || exitclean
+      cp_p $CDMNT/LiveOS/squashfs.img $USBMNT/$LIVEOS/squashfs.img || exitclean
   elif [ -f $CDMNT/LiveOS/ext3fs.img ]; then
       cp $CDMNT/LiveOS/ext3fs.img $USBMNT/$LIVEOS/ext3fs.img || exitclean
   fi
