diff -Naur livecd-tools-18.8/imgcreate/live.py livecd-tools-18.8.tpg/imgcreate/live.py
--- livecd-tools-18.8/imgcreate/live.py	2014-03-16 16:36:04.000000000 +0000
+++ livecd-tools-18.8.tpg/imgcreate/live.py	2014-03-16 16:42:57.918007585 +0000
@@ -458,8 +458,8 @@
                         preexec_fn = self._chroot)
 
         isDracut = False
-        if os.path.exists(bootdir + "/initramfs-" + version + ".img"):
-            shutil.copyfile(bootdir + "/initramfs-" + version + ".img",
+        if os.path.exists(bootdir + "/liveramfs-" + version + ".img"):
+            shutil.copyfile(bootdir + "/liveramfs-" + version + ".img",
                             isodir + "/isolinux/initrd" + index + ".img")
             isDracut = True
         else:
