--- livecd-tools-18.8/imgcreate/live.py	2014-11-08 01:21:59.707954648 +0000
+++ livecd-tools-18.8/imgcreate/live.py~	2014-11-10 14:04:02.951666036 +0000
@@ -754,10 +771,6 @@
 
     def __get_basic_efi_config(self, **args):
         return """
-default=0
-timeout %(timeout)d
-hiddenmenu
-set default="0"
 
 function load_video {
   insmod efi_gop
@@ -786,9 +799,10 @@
             args["rootlabel"] = "live:LABEL=%(fslabel)s" % args
         else:
             args["rootlabel"] = "CDLABEL=%(fslabel)s" % args
-        return """title %(long)s
-  kernel /isolinux/vmlinuz%(index)s root=%(rootlabel)s rootfstype=%(isofstype)s %(liveargs)s %(extra)s
+        return """echo %(long)s
+  linux /isolinux/vmlinuz%(index)s root=%(rootlabel)s rootfstype=%(isofstype)s %(liveargs)s %(extra)s
   initrd /isolinux/initrd%(index)s.img
+  boot
 """ %args
 
     def __get_efi_image_stanzas(self, isodir, name):
@@ -827,24 +841,24 @@
             shutil.rmtree(isodir + "/EFI")
             return
 
-        cfg = self.__get_basic_efi_config(name = self.name,
+        cfg = self.__get_basic_efi_config(isolabel = self.fslabel,
                                           timeout = self._timeout)
         cfg += self.__get_efi_image_stanzas(isodir, self.name)
 
-        cfgf = open(isodir + "/EFI/BOOT/grub.conf", "w")
+        cfgf = open(isodir + "/EFI/BOOT/grub.cfg", "w")
         cfgf.write(cfg)
         cfgf.close()
 
         # first gen mactel machines get the bootloader name wrong apparently
         if getBaseArch() == "i386":
             os.link(isodir + "/EFI/BOOT/grub.efi", isodir + "/EFI/BOOT/BOOT.efi")
-            os.link(isodir + "/EFI/BOOT/grub.conf", isodir + "/EFI/BOOT/BOOT.conf")
+            os.link(isodir + "/EFI/BOOT/grub.cfg", isodir + "/EFI/BOOT/BOOT.cfg")
 
         # for most things, we want them named boot$efiarch
         efiarch = {"i386": "IA32", "x86_64": "X64"}
         efiname = efiarch[getBaseArch()]
         os.rename(isodir + "/EFI/BOOT/grub.efi", isodir + "/EFI/BOOT/BOOT%s.efi" %(efiname,))
-        os.link(isodir + "/EFI/BOOT/grub.conf", isodir + "/EFI/BOOT/BOOT%s.conf" %(efiname,))
+        os.link(isodir + "/EFI/BOOT/grub.cfg", isodir + "/EFI/BOOT/BOOT%s.cfg" %(efiname,))
 
 
     def _configure_bootloader(self, isodir):
