--- livecd-tools-18.8/imgcreate/live.py~	2014-12-22 00:23:43.739149030 +0000
+++ livecd-tools-18.8/imgcreate/live.py	2014-12-22 00:26:02.838143383 +0000
@@ -816,15 +816,15 @@
             args["rootlabel"] = "live:LABEL=%(fslabel)s" % args
         else:
             args["rootlabel"] = "CDLABEL=%(fslabel)s" % args
-        return """menuentry 'Start  %(long)s' --id 'linux0' {
+        return """menuentry 'Start OpenMandrivaLx-2014.1 ' --id 'linux0' {
   linux /isolinux/vmlinuz%(index)s root=%(rootlabel)s rootfstype=%(isofstype)s %(liveargs)s %(extra)s rd.luks=0 rd.md=0 rd.dm=0
   initrd /isolinux/initrd%(index)s.img
   }
-  menuentry 'Install %(long)s' --id 'linux0' {
+  menuentry 'Install OpenMandrivaLx-2014.1' --id 'linux0' {
   linux /isolinux/vmlinuz%(index)s root=%(rootlabel)s rootfstype=%(isofstype)s %(liveargs)s %(extra)s install
   initrd /isolinux/initrd%(index)s.img
   }
-  menuentry 'Install %(long)s in basic graphics mode' --id 'basic0' {
+  menuentry 'Install OpenMandrivaLx-2014.1 in basic graphics mode' --id 'basic0' {
   linux /isolinux/vmlinuz%(index)s root=%(rootlabel)s rootfstype=%(isofstype)s %(liveargs)s %(extra)s install
   initrd /isolinux/initrd%(index)s.img
   }
