diff -ur livecd-tools-031.orig/imgcreate/creator.py livecd-tools-031/imgcreate/creator.py
--- livecd-tools-031.orig/imgcreate/creator.py	2010-11-29 17:53:37.000000000 +0300
+++ livecd-tools-031/imgcreate/creator.py	2010-11-29 17:53:57.233005234 +0300
@@ -740,7 +740,7 @@
         """
         ksh = self.ks.handler
 
-        kickstart.LanguageConfig(self._instroot).apply(ksh.lang)
+        #kickstart.LanguageConfig(self._instroot).apply(ksh.lang)
         kickstart.KeyboardConfig(self._instroot).apply(ksh.keyboard)
         kickstart.TimezoneConfig(self._instroot).apply(ksh.timezone)
         kickstart.AuthConfig(self._instroot).apply(ksh.authconfig)
Только в livecd-tools-031/imgcreate: creator.py~
