--- livecd-tools-18.8/imgcreate/live.py.orig	2013-11-01 11:12:13.881951711 +0000
+++ livecd-tools-18.8/imgcreate/live.py	2013-11-01 11:12:41.752953177 +0000
@@ -459,8 +459,6 @@
 
         isDracut = False
         if os.path.exists(bootdir + "/initramfs-" + version + ".img"):
-            shutil.copyfile(bootdir + "/initramfs-" + version + ".img",
-                            isodir + "/isolinux/initrd" + index + ".img")
             isDracut = True
         else:
             raise InitramfsError("Unable to create initramfs-" + version + ".img")
