diff -ur livecd-tools-031.orig/imgcreate/creator.py livecd-tools-031/imgcreate/creator.py
--- livecd-tools-031.orig/imgcreate/creator.py	2010-10-05 11:13:26.000000000 +0400
+++ livecd-tools-031/imgcreate/creator.py	2010-10-05 11:23:00.259235979 +0400
@@ -16,6 +16,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
+import time
 import os
 import os.path
 import stat
@@ -623,6 +624,23 @@
         if not kickstart.selinux_enabled(self.ks) and os.path.exists("/selinux/enforce") and not ayum.installHasFile(file):
             raise CreatorError("Unable to disable SELinux because the installed package set did not include the file %s" % (file))
 
+    def install_urpmi(self, repo_urls = {}):
+      
+        #urpmi_conf = self._instroot + "/urpmi-conf" 
+        urpmi_conf = self.__builddir + "/urpmi_conf"
+        print urpmi_conf
+        time.sleep(5)
+
+        for repo in kickstart.get_repos(self.ks, repo_urls):
+            (name, baseurl, mirrorlist, inc, exc) = repo
+            subprocess.call(["/usr/sbin/urpmi.addmedia", "--urpmi-root", urpmi_conf, name, baseurl])
+
+        packages = self.ks.handler.packages.packageList
+
+        #for package in self.ks.handler.packages.packageList:
+            #subprocess.call(["/usr/sbin/urpmi", "--auto", "--urpmi-root", urpmi_conf, "--root", self._instroot, package])
+        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
+
     def install(self, repo_urls = {}):
         """Install packages into the install root.
 
Только в livecd-tools-031/imgcreate: creator.py~
diff -ur livecd-tools-031.orig/imgcreate/live.py livecd-tools-031/imgcreate/live.py
--- livecd-tools-031.orig/imgcreate/live.py	2010-10-05 11:13:26.000000000 +0400
+++ livecd-tools-031/imgcreate/live.py	2010-10-05 11:19:59.971988213 +0400
@@ -478,7 +478,7 @@
         checkisomd5 = self._has_checkisomd5()
 
         cfg = ""
-
+        print "++++++++++++++++++++++++", versions
         index = "0"
         for version in versions:
             (is_xen, isDracut) = self.__copy_kernel_and_initramfs(isodir, version, index)
diff -ur livecd-tools-031.orig/tools/livecd-creator livecd-tools-031/tools/livecd-creator
--- livecd-tools-031.orig/tools/livecd-creator	2010-10-05 11:13:26.000000000 +0400
+++ livecd-tools-031/tools/livecd-creator	2010-10-05 11:19:59.971988213 +0400
@@ -129,7 +129,7 @@
 
     try:
         creator.mount(options.base_on, options.cachedir)
-        creator.install()
+        creator.install_urpmi()
         creator.configure()
         if options.give_shell:
             print "Launching shell. Exit to continue."
