diff -Nur livecd-tools-18.8.orig/imgcreate/live.py livecd-tools-18.8/imgcreate/live.py
--- livecd-tools-18.8.orig/imgcreate/live.py	2012-08-06 22:49:20.000000000 +0400
+++ livecd-tools-18.8/imgcreate/live.py	2013-04-24 18:37:44.591687049 +0400
@@ -254,7 +254,7 @@
         return env
 
     def __extra_filesystems(self):
-        return "vfat msdos ";
+        return "vfat msdos =fs/nls ";
 
     def __extra_drivers(self):
         retval = "sr_mod sd_mod ide-cd cdrom "
