diff -Naur livecd-tools-18.8/imgcreate/creator.py livecd-tools-18.8.tpg/imgcreate/creator.py
--- livecd-tools-18.8/imgcreate/creator.py	2014-12-07 13:08:41.000000000 +0000
+++ livecd-tools-18.8.tpg/imgcreate/creator.py	2014-12-07 13:21:44.965999684 +0000
@@ -620,10 +620,19 @@
             subprocess.call(["/usr/sbin/urpmi.addmedia", "--urpmi-root", urpmi_conf, name, baseurl])
 
         packages = self.ks.handler.packages.packageList
+	if "basesystem" in packages:
+            print "Basesystem should be installed first"
+            if "kernel-nrjQL-desktop-latest" in packages:
+                subprocess.call(["/usr/sbin/urpmi", "--auto", "--split-length", "0", "--fastunsafe", "--nolock", "--ignorearch", "--no-verify-rpm", "--no-suggests", "--ignoresize", "--urpmi-root", urpmi_conf, "--root", self._instroot] + ['basesystem', 'kernel-nrjQL-desktop-latest'])
+            else:
+                subprocess.call(["/usr/sbin/urpmi", "--auto", "--split-length", "0", "--fastunsafe", "--nolock", "--ignorearch", "--no-verify-rpm", "--no-suggests", "--ignoresize", "--urpmi-root", urpmi_conf, "--root", self._instroot] + ['basesystem'])
+
+        print "Now let's install all other packages"
+        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--fastunsafe", "--debug", "--no-verify-rpm", "--ignoresize", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
 
         #for package in self.ks.handler.packages.packageList:
             #subprocess.call(["/usr/sbin/urpmi", "--auto", "--urpmi-root", urpmi_conf, "--root", self._instroot, package])
-        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--fastunsafe", "--ignoresize", "--no-verify", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
+#        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--fastunsafe","--no-verify", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
 
 
 #    def install(self, repo_urls = {}):
