diff -Nur livecd-tools-031.orig/docs/livecd-iso-to-disk.pod livecd-tools-031/docs/livecd-iso-to-disk.pod
--- livecd-tools-031.orig/docs/livecd-iso-to-disk.pod	2009-10-30 00:26:09.000000000 +0300
+++ livecd-tools-031/docs/livecd-iso-to-disk.pod	2011-02-16 16:47:46.427623627 +0300
@@ -6,45 +6,21 @@
 
 =head1 SYNOPSIS
 
-B<livecd-iso-to-disk> [--format] [--reset-mbr] [--noverify] [--overlay-size-mb <size>] [\-\-home\-size\-mb <size>] [\-\-unencrypted\-home] [\-\-skipcopy]  <path-to-iso> <usb storage device>
+B<livecd-iso-to-disk> [--verify] <path-to-iso> <usb storage device>
 
 =head1 DESCRIPTION
 
 B<livecd-iso-to-disk> installs a live CD or DVD image(ISO image) onto a USB storage device. The USB storage device can then boot the installed operating system on systems which support booting via USB. B<livecd-iso-to-disk> requires an ISO image and a USB storage device. ISO images can be created with B<livecd-creator>.
 
-B<livecd-iso-to-disk> is not a destructive process; any data you currently have on your USB stick is preserved.
+B<livecd-iso-to-disk> is a destructive process; any data you currently have on your USB stick is lost.
 
 =head1 OPTIONS
 
 =over 4
 
-=item --format
+=item --verify
 
-Formats the USB stick and creates a GPT partition table.
-
-=item --reset-mbr
-
-Sets the Master Boot Record(MBR) of the USB storage device to the mbr.bin file from the image's syslinux directory.
-
-=item --noverify
-
-Disables the image validation process which occurs before the image is installed. When this option is enabled the image is not verified before installation on the USB storage device.
-
-=item --overlay-size-mb
-
-This option sets the overlay size in megabytes. The overlay is additional storage available to the live operating system if the operating system supports it. The USB storage device must have enough free space for the image and the overlay.
-
-=item --home-size-mb
-
-Sets the home directory size in megabytes.
-
-=item --unencrypted-home
-
-Disables the encryption of the home directory.
-
-=item --skipcopy
-
-Skips the copy of the live image to the USB stick.
+Enables the image validation process which occurs before the image is installed. When this option is enabled the image is not verified before installation on the USB storage device.
 
 =back
 
