diff -Nur livecd-tools-031.orig/tools/livecd-iso-to-disk-oem.sh livecd-tools-031/tools/livecd-iso-to-disk-oem.sh
--- livecd-tools-031.orig/tools/livecd-iso-to-disk-oem.sh	2011-04-05 11:43:52.000000000 +0400
+++ livecd-tools-031/tools/livecd-iso-to-disk-oem.sh	2011-04-05 12:53:51.752192945 +0400
@@ -295,7 +295,7 @@
 createpartitions () {
 
     #cyl_size=$(/usr/sbin/parted $device -sm unit cyl p | awk {'FS=":"; if (FNR==3) print $4'}| sed -e 's/kB;$//')
-    cyl_size=$(( $(/sbin/fdisk -l $device | awk '{ if (FNR==4) print $9 }')/1024 ))
+    cyl_size=$(( $(/sbin/fdisk -u=cylinders -l $device | awk '{ if (FNR==4) print $9 }')/1024 ))
 
     devinfo=$(/usr/sbin/parted --script -m $device "unit MB print" |grep ^$device:)
     devsize=$(echo $devinfo |cut -d : -f 2 |sed -e 's/MB$//')
@@ -311,7 +311,7 @@
     if [ "$systemsize" -le "$devsize" ]
         then
 
-        /sbin/fdisk $device 2>/dev/null 1>/dev/null << EOF
+        /sbin/fdisk -u=cylinders $device 2>/dev/null 1>/dev/null << EOF
         o
         n
         p
