diff -ur livecd-tools-031.orig/tools/livecd-creator livecd-tools-031/tools/livecd-creator
--- livecd-tools-031.orig/tools/livecd-creator	2010-11-12 14:12:28.000000000 +0300
+++ livecd-tools-031/tools/livecd-creator	2010-11-12 14:21:39.517648703 +0300
@@ -43,6 +43,8 @@
                       help="File system label (default based on config name)")
     imgopt.add_option("-m", "--menu-config", type="string", dest="menu_cfg",
 		      help="Extra options for LiveCD menu")
+    imgopt.add_option("", "--yum", action="store_true", dest="yum",
+		      help="Use yum instead of urpmi")
 
     parser.add_option_group(imgopt)
 
@@ -129,7 +131,10 @@
 
     try:
         creator.mount(options.base_on, options.cachedir)
-        creator.install_urpmi()
+        if options.yum:
+            creator.install()
+        else:
+            creator.install_urpmi()
         creator.configure()
         if options.give_shell:
             print "Launching shell. Exit to continue."
Только в livecd-tools-031/tools: livecd-creator~
