diff -Naur livecd-tools-18.8/imgcreate/kickstart.py livecd-tools-18.8.tpg/imgcreate/kickstart.py
--- livecd-tools-18.8/imgcreate/kickstart.py	2014-03-28 21:17:47.000000000 +0000
+++ livecd-tools-18.8.tpg/imgcreate/kickstart.py	2014-03-28 21:20:50.822214841 +0000
@@ -131,20 +131,28 @@
     def apply(self, kslang):
         lang = kslang.lang or "en_US.UTF-8"
 
-        f = open(self.path("/etc/sysconfig/i18n"), "w+")
+        f = open(self.path("/etc/locale.conf"), "w+")
         f.write("LANG=\"" + lang + "\"\n")
         f.close()
 
 class KeyboardConfig(KickstartConfig):
     """A class to apply a kickstart keyboard configuration to a system."""
     def apply(self, kskeyboard):
-#        k = keyboard.Keyboard()
-            f = open(self.path("/etc/sysconfig/keyboard"), "w+")
-            f.write("KEYBOARD=\"" + kskeyboard.keyboard + "\"\n")
-            f.close()
-#        if kskeyboard.keyboard:
-#            k.set(kskeyboard.keyboard)
-#        k.write(self.instroot)
+        vcconf_file = self.path("/etc/vconsole.conf")
+        DEFAULT_VC_FONT = "latarcyrheb-sun16"
+
+        if not kskeyboard.keyboard:
+            kskeyboard.keyboard = "us"
+
+        try:
+            with open(vcconf_file, "w") as f:
+                f.write('KEYMAP="%s"\n' % kskeyboard.keyboard)
+
+                # systemd now defaults to a font that cannot display non-ascii
+                # characters, so we have to tell it to use a better one
+                f.write('FONT="%s"\n' % DEFAULT_VC_FONT)
+        except IOError as e:
+            logging.error("Cannot write vconsole configuration file: %s" % e)
 
 class TimezoneConfig(KickstartConfig):
     """A class to apply a kickstart timezone configuration to a system."""
