--- livecd-tools-031/imgcreate/creator.py.no-verify	2011-03-01 10:30:23.000000000 +0100
+++ livecd-tools-031/imgcreate/creator.py	2011-03-01 10:36:22.000000000 +0100
@@ -640,10 +640,10 @@
 
         #for package in self.ks.handler.packages.packageList:
             #subprocess.call(["/usr/sbin/urpmi", "--auto", "--urpmi-root", urpmi_conf, "--root", self._instroot, package])
-        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
+        subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-verify-rpm", "--no-suggests", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages)
 
         if packages_post:
-            subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-suggests", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages_post)
+            subprocess.call(["/usr/sbin/urpmi", "--auto", "--no-verify-rpm", "--no-suggests", "--urpmi-root", urpmi_conf, "--root", self._instroot] + packages_post)
 
     #def install(self, repo_urls = {}):
         #"""Install packages into the install root.
