--- livecd-tools-031/imgcreate/fs.py.xz	2011-02-11 13:39:10.000000000 -0200
+++ livecd-tools-031/imgcreate/fs.py	2011-02-11 13:39:16.000000000 -0200
@@ -42,7 +42,7 @@ def makedirs(dirname):
 
 def mksquashfs(in_img, out_img):
     #args = ["/sbin/mksquashfs", in_img, out_img]
-    args = ["/usr/bin/mksquashfs", in_img, out_img, "-comp", "lzma"]
+    args = ["/usr/bin/mksquashfs", in_img, out_img, "-comp", "xz"]
 
     if not sys.stdout.isatty():
         args.append("-no-progress")
